### ISSUE TITLE
subscriptionError Connection failed ... Missing field argument owner when using OIDC

### ISSUE THREAD

**Issue by getglad** [aws-amplify/amplify-js#6592]
#**Describe the bug**
When using `@auth (rules: [  { allow: owner, provider: oidc, identityClaim: "sub" }  ])`, receive `subscriptionError Connection failed ... Missing field argument owner` error message.

The reported error is similar to an ready reported bug (https://github.com/aws-amplify/amplify-js/issues/6108), but ultimately I believe my root cause is different, thus the fix is different.

See fix under TLDR at the bottom of all of this

**To Reproduce**
Add the above @auth rule and run an application (in my case it is an Angular 9 app)

**Additional context**
So on the last commit for `amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts`, it looks like `subscriptionError` got added to the error message, which has let me back into the problem a bit.

This error gets thrown here: https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts#L354-L367

Poking around in that area, I noticed something in https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts#L302-L312

If the evaluation of both `if (isOwner)` and `if (!ownerValue)` pass, it throws the error message of `Owner field required, sign in is needed in order to perform this operation` - and the subscription is never attempted because we return at that step.

In my case, `isOwner` comes through as `undefined`. This means we continue to make use of `API.graphql`, which will fail because there is no `ownerField` added to `variables` - which means we eventually land in the error node of the `subscribe`. 

So this leads me to wonder what steps get us to the point where both `isOwner` and `ownerValue` are `undefined`?

https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts#L280-L287
https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts#L77
https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts#L61-L68

Best I can tell, this is coming from the stored return of `buildSubscription` in `queriesMetadata`. `buildSubscription` gets `isOwner` and `ownerValue` from `getAuthorizationInfo`. Using the existing unit tests, I can tell this should not be returning `undefined` if the inputs for `oidcTokenPayload` are correct, so I looked at where `oidcTokenPayload` is referenced from.

*TLDR:*

https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts#L258-L269

Best I can tell,  `oidcTokenPayload` comes from the above code. But that is where I notice it is using `Buffer` and swallowing (not even logging) any error. When I try to just execute the lines myself using a token, I get a `Buffer is not defined` error.

To fix:
1) Running `npm install buffer`
2) adding `(window as any).global.Buffer = (window as any).global.Buffer || require('buffer').Buffer;` to `pollyfills.ts`
3) Save/Change some data through the app
4) Use query interface in app sync, see it reflected there

This all could probably have been avoided with a reference for Buffer here: https://docs.amplify.aws/start/getting-started/setup/q/integration/angular#create-a-new-angular-app - but you guys may have a better idea that would be more universal

**Comment by mauerbac** [comment]
hi @getglad - thanks for using datastore and taking the time to provide detailed reports. This seems like a duplicate of the problem you described [here](https://github.com/aws-amplify/amplify-js/issues/6592). I will close this issue. 

The team will use the other issue to discuss a proper solution.

**Comment by getglad** [comment]
I would probably argue that this is actually a different issue involving the use of Buffer - different from read/write location of the token - related sure but even if the token storage were to be fixed the Buffer invocation would fail
